pytorch_lightning/trainer/evaluation_loop.py

```diff
@@ -136,6 +140,11 @@ def setup(self, model, max_batches, dataloaders):
         self.max_batches = max_batches
         self.num_dataloaders = self._get_num_dataloaders(dataloaders)
 
+        # wrap user samplers, so we can capture batch indices
+        dataloaders = [LightningBatchSamplerWrapper.to_new_dataloader(dataloader) for dataloader in dataloaders]
+        batch_samplers = [getattr(dataloader, "batch_sampler", None) for dataloader in dataloaders]
```

Comment:
Can these be None?

Reply:
You are right, I don't think so.

Reply:
```
        if batch_size is not None and batch_sampler is None:
            # auto_collation without custom batch_sampler
            batch_sampler = BatchSampler(sampler, batch_size, drop_last)
```

Yes, it should always be defined even with batch_size = 1. Removing the getattr